File: Modules/ServerTypes/AnnualLeaveJobGradePolicy/AnnualLeaveJobGradePolicyForm.ts
import { LookupEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnualLeaveJobGradePolicyForm {
    JobGradeLevel: LookupEditor;
    EligibleDays: IntegerEditor;
    MaximumAccumulated: IntegerEditor;
}

export class AnnualLeaveJobGradePolicyForm extends PrefixedContext {
    static formKey = 'AnnualLeaveJobGradePolicy.AnnualLeaveJobGradePolicy';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnualLeaveJobGradePolicyForm.init)  {
            AnnualLeaveJobGradePolicyForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;

            initFormType(AnnualLeaveJobGradePolicyForm, [
                'JobGradeLevel', w0,
                'EligibleDays', w1,
                'MaximumAccumulated', w1
            ]);
        }
    }
}

File: Modules/HumanResource/AnnualLeaveJobGradePolicy/AnnualLeaveJobGradePolicy/AnnualLeaveJobGradePolicyForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.AnnualLeaveJobGradePolicy.Forms;

[FormScript("AnnualLeaveJobGradePolicy.AnnualLeaveJobGradePolicy")]
[BasedOnRow(typeof(AnnualLeaveJobGradePolicyRow), CheckNames = true)]
public class AnnualLeaveJobGradePolicyForm
{
    public int JobGradeLevel { get; set; }
    public int EligibleDays { get; set; }
    public int MaximumAccumulated { get; set; }
}